app/models/detailed_guide.rb

```diff
@@ -62,6 +65,35 @@ def related_detailed_guide_ids=(detailed_guide_ids)
     self.related_documents = other_related_documents + detailed_guide_documents
   end
 
+  def need_ids
+    return @need_ids if @need_ids
+    return [] unless document # TODO: Remove
+
+    response = Whitehall.publishing_api_v2_client.get_links(
+      document.content_id
+    )
+
+    return unless response
+
+    @need_ids = response["links"]["meets_user_needs"]
+  end
+
+  def patch_meets_user_needs_links
+    return unless @need_ids
```

Comment:
I think the current code does manage, though this does require the rather ugly `return if @need_ids.is_a? String`. If the `need_ids` column is removed from Whitehall, this can then be removed.